src/qtgui/ioconfig.cpp

```diff
@@ -254,12 +256,20 @@ void CIoConfig::saveConfig()
 
     idx = ui->outDevCombo->currentIndex();
 
-    if (idx > 0)
+    if (idx > 0
```

Comment:
I would prefer not putting an #ifdef inside an expression. Perhaps something like this insted:

```
#if defined(WITH_PULSEAUDIO) || defined(GQRX_OS_MACX)
    if (idx > 0)
    {
        qDebug() << "Output device" << idx << ":" << QString(outDevList[idx-1].get_name().c_str());
        m_settings->setValue("output/device", QString(outDevList[idx-1].get_name().c_str()));
    }
#else
    if (idx > 0 || ui->outDevCombo->currentText() != "Default")
    {
        qDebug() << "Output device:" << ui->outDevCombo->currentText();
        m_settings->setValue("output/device", ui->outDevCombo->currentText());
    }
#endif
    else
    {
        m_settings->remove("output/device");
    }

```